src/main/java/io/aiven/kafka/connect/http/sender/TrustStoreLoader.java

```diff
@@ -81,16 +81,26 @@ private static InputStream tryFileSystem(final String path) {
             return null;
         }
         try {
-            final URL jarLocation = TrustStoreLoader.class.getProtectionDomain().getCodeSource().getLocation();
-            LOG.debug("JAR location: {}", jarLocation);
-            final Path jarPath = Paths.get(jarLocation.toURI());
-            final Path parentPath = jarPath.getParent();
-            if (parentPath == null) {
-                LOG.info("JAR has no parent directory, skipping file system lookup");
-                return null;
+            final Path truststorePath;
+            if (Paths.get(path).isAbsolute()) {
+                // Use absolute path directly
+                truststorePath = Paths.get(path);
+                LOG.info("Using absolute file system path: {}", truststorePath);
+            } else {
+                // Resolve relative path against JAR location
+                final URL jarLocation = TrustStoreLoader.class.getProtectionDomain().getCodeSource().getLocation();
+                LOG.info("JAR location: {}", jarLocation);
+                final Path jarPath = Paths.get(jarLocation.toURI());
+                LOG.info("Using relative path against the JAR locataion: {}", jarPath);
+                final Path parentPath = jarPath.getParent();
+                LOG.info("JAR parentPath: {}", parentPath);
+                if (parentPath == null) {
+                    LOG.info("JAR has no parent directory, skipping file system lookup");
+                    return null;
+                }
+                truststorePath = parentPath.resolve(path.startsWith("/") ? path.substring(1) : path);
```

Comment:
```suggestion
                truststorePath = parentPath.resolve(path);
```

IIUC, it isn't possible that the path starts with "/" because of the `isAbsolute()` call.